services/analytics-service-base.ts

```diff
@@ -134,29 +135,92 @@ export class AnalyticsServiceBase implements IAnalyticsService {
 		return this.getHumanReadableStatusMessage(settingName, readableSettingName);
 	}
 
-	public async restartEqatecMonitor(projectApiKey: string): Promise<void> {
-		this.tryStopEqatecMonitor();
-		await this.start(projectApiKey);
-	}
-
-	protected checkConsentCore(trackFeatureUsage: boolean): Promise<void> {
-		return this.trackFeatureCore(`${this.acceptTrackFeatureSetting}.${!!trackFeatureUsage}`);
-	}
-
 	protected async trackFeatureCore(featureTrackString: string): Promise<void> {
 		try {
 			if (await this.$analyticsSettingsService.canDoRequest()) {
-				await this.start();
-				if (this._eqatecMonitor) {
-					this._eqatecMonitor.trackFeature(featureTrackString);
-					await this.waitForSending();
-				}
+				await this.sendDataToEqatecMonitors(this.featureTrackingAPIKeys, (eqatecMonitor: IEqatecMonitor) => eqatecMonitor.trackFeature(featureTrackString));
 			}
 		} catch (e) {
-			this.$logger.trace("Analytics exception: '%s'", e.toString());
+			this.$logger.trace(`Analytics exception: '${e}'`);
 		}
 	}
 
+	@cache()
+	protected async initAnalyticsStatuses(): Promise<void> {
+		if (await this.$analyticsSettingsService.canDoRequest()) {
+			this.$logger.trace("Initializing analytics statuses.");
+			const settingsNames = [this.$staticConfig.TRACK_FEATURE_USAGE_SETTING_NAME, this.$staticConfig.ERROR_REPORT_SETTING_NAME];
+
+			for (const settingName of settingsNames) {
+				await this.getStatus(settingName);
+			}
+
+			this.$logger.trace("Analytics statuses: ");
+			this.$logger.trace(this.analyticsStatuses);
+		}
+	}
+
+	private getIsSending(eqatecMonitor: any): boolean {
+		return eqatecMonitor.status().isSending;
+	}
+
+	private waitForSending(eqatecMonitor: any): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			const intervalTime = 100;
+			let remainingTime = AnalyticsServiceBase.MAX_WAIT_SENDING_INTERVAL;
+
+			if (this.getIsSending(eqatecMonitor)) {
+				this.$logger.trace(`Waiting for analytics to send information. Will check in ${intervalTime}ms.`);
+				const interval = setInterval(() => {
+					if (!this.getIsSending(eqatecMonitor) || remainingTime <= 0) {
+						clearInterval(interval);
+						resolve();
+					}
+
+					remainingTime -= intervalTime;
+					this.$logger.trace(`Waiting for analytics to send information. Will check in ${intervalTime}ms. Remaining time is: ${remainingTime}`);
```

Comment:
The first part of this message is duplicated in a `trace` above - can be extracted as a `const` variable